### PR TITLE
[datadog_integration_azure] Add resource_collection_enabled

### PR DESCRIPTION
**Pull request by ava-silver**
Adds Resource Collection field (`resource_collection_enabled`) to the Azure Integration.

The new field is optional with no default, as it interacts with `cspm_enabled` so we want to let the API dictate the behavior when not specified by not providing a default value here.

### CODE REVIEW

File: docs/resources/dashboard.md
Comment (ava-silver) [inline docs/resources/dashboard.md:1]: autogenerated by `make docs`, not my change

File: docs/resources/integration_azure.md [hunk 1]
```diff
@@ -41,9 +41,11 @@ resource "datadog_integration_azure" "sandbox" {
...
 - `automute` (Boolean) Silence monitors for expected Azure VM shutdowns. Defaults to `false`.
 - `container_app_filters` (String) This comma-separated list of tags (in the form `key:value,key:value`) defines a filter that Datadog uses when collecting metrics from Azure Container Apps. Only Container Apps that match one of the defined tags are imported into Datadog. Defaults to `""`.
-- `cspm_enabled` (Boolean) Enable Cloud Security Management Misconfigurations for your organization. Defaults to `false`.
+- `cspm_enabled` (Boolean) When enabled, Datadog’s Cloud Security Management product will scan resource configurations monitored by this app registration.
+Note: This requires `resource_collection_enabled` to be set to true. Defaults to `false`.
```
Comment (urseberry) [inline docs/resources/integration_azure.md:45]:
```suggestion
- `cspm_enabled` (Boolean) When enabled, Datadog’s Cloud Security Management product scans resource configurations monitored by this app registration.
**Note:** This parameter requires `resource_collection_enabled` to be set to true. `resource_collection_enabled` defaults to `false`, so change it to `true` if you enable `cspm_enabled`.
```

Comment (ava-silver) [reply docs/resources/integration_azure.md:45]: the `Defaults to \`false\`.` part is autogenerated at the end, however the `resource_collection_enabled` is optional, the only constraint set by the API is that if `cspm_enabled` is `true`, then `resource_collection_enabled` cannot be explicitly set to `false`.

would something like the following work?
```md
**Note:** This parameter cannot be set to true if `resource_collection_enabled` is explicitly set to false, as this feature depends on resource collection. Defaults to `false`.
```

File: docs/resources/integration_azure.md [hunk 2]
```diff
@@ -41,9 +41,11 @@ resource "datadog_integration_azure" "sandbox" {
 - `app_service_plan_filters` (String) This comma-separated list of tags (in the form `key:value,key:value`) defines a filter that Datadog uses when collecting metrics from Azure App Service Plans. Only App Service Plans that match one of the defined tags are imported into Datadog. The rest, including the apps and functions running on them, are ignored. This also filters the metrics for any App or Function running on the App Service Plan(s). Defaults to `""`.
 - `automute` (Boolean) Silence monitors for expected Azure VM shutdowns. Defaults to `false`.
 - `container_app_filters` (String) This comma-separated list of tags (in the form `key:value,key:value`) defines a filter that Datadog uses when collecting metrics from Azure Container Apps. Only Container Apps that match one of the defined tags are imported into Datadog. Defaults to `""`.
-- `cspm_enabled` (Boolean) Enable Cloud Security Management Misconfigurations for your organization. Defaults to `false`.
+- `cspm_enabled` (Boolean) When enabled, Datadog’s Cloud Security Management product will scan resource configurations monitored by this app registration.
```
Comment (urseberry) [inline docs/resources/integration_azure.md:44]:
```suggestion
- `cspm_enabled` (Boolean) When enabled, Datadog’s Cloud Security Management product scans resource configurations monitored by this app registration.
```
Updated suggestion based on our conversation.

**Review by urseberry — APPROVED** [review]

**Review by parsons90 — APPROVED** [review]

**Review by skarimo — APPROVED** [review]

**Pull request merged**